File: backend/app/schemas/family.py
from pydantic import BaseModel
from uuid import UUID
from datetime import date, datetime
from typing import Optional


class FamilyMemberCreate(BaseModel):
    full_name: str
    relationship: str
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    photo_url: Optional[str] = None
    blood_group: Optional[str] = None
    guardian_name: Optional[str] = None
    guardian_phone: Optional[str] = None
    notes: Optional[str] = None


class FamilyMemberUpdate(BaseModel):
    full_name: Optional[str] = None
    relationship: Optional[str] = None
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    photo_url: Optional[str] = None
    blood_group: Optional[str] = None
    guardian_name: Optional[str] = None
    guardian_phone: Optional[str] = None
    notes: Optional[str] = None


class FamilyMemberResponse(BaseModel):
    id: UUID
    user_id: UUID
    full_name: str
    relationship: str
    date_of_birth: Optional[date] = None
    gender: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    photo_url: Optional[str] = None
    blood_group: Optional[str] = None
    guardian_name: Optional[str] = None
    guardian_phone: Optional[str] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: mobile/src/types/family.ts
export interface FamilyMember {
  id: string;
  user_id: string;
  full_name: string;
  relationship: string;
  date_of_birth: string | null;
  gender: string | null;
  phone: string | null;
  email: string | null;
  photo_url: string | null;
  blood_group: string | null;
  guardian_name: string | null;
  guardian_phone: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface FamilyMemberCreate {
  full_name: string;
  relationship: string;
  date_of_birth?: string | null;
  gender?: string | null;
  phone?: string | null;
  email?: string | null;
  photo_url?: string | null;
  blood_group?: string | null;
  guardian_name?: string | null;
  guardian_phone?: string | null;
  notes?: string | null;
}

export type FamilyMemberUpdate = Partial<FamilyMemberCreate>;
